resolver/resolver.go

```diff
@@ -319,3 +319,12 @@ type Resolver interface {
 	// Close closes the resolver.
 	Close()
 }
+
+// AuthorityOverrider is implemented by Builders that wish to override the
+// default authority for the ClientConn.
+// By default, the authority used is target.Endpoint()
+type AuthorityOverrider interface {
+	// GetServiceAuthority returns the authority to use for a ClientConn with the
+	// given target. It must not perform I/O or any other blocking operations.
+	GetServiceAuthority(t Target) string
```

Comment:
Go style generally doesn't like `GetXYZ`.  `OverrideAuthority` makes more sense given the name of the interface; otherwise `AuthorityForTarget` or something like that could be okay.